libs/vkd3d/device.c

```diff
@@ -8046,7 +8053,7 @@ static void d3d12_device_caps_init_feature_options16(struct d3d12_device *device
     D3D12_FEATURE_DATA_D3D12_OPTIONS16 *options16 = &device->d3d12_caps.options16;
 
     options16->DynamicDepthBiasSupported = TRUE;
-    options16->GPUUploadHeapSupported = FALSE;
+    options16->GPUUploadHeapSupported = device->memory_info.rebar_budget != 0;
```

Comment:
This is conflicting with

```
    if (heap_properties->MemoryPoolPreference == D3D12_MEMORY_POOL_L1
        && heap_properties->CPUPageProperty == D3D12_CPU_PAGE_PROPERTY_WRITE_COMBINE
        && !(device->memory_info.upload_heap_memory_properties & VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT))
    {
        WARN("Invalid memory pool preference (device does not support rebar).\n");
        return E_INVALIDARG;
    }
```

which uses the upload_heap_memory_properties instead to determine if GPUUploadHeap is supported.

Reply:
I think we might want to use vkd3d_memory_topology_is_uma_like() check in vkd3d_memory_info_upload_hvv_memory_properties() somehow.

Perhaps the cleaner solution is to not request DEVICE_LOCAL properties for UMA-like, since device-local and host-visible is kinda the same thing anyway, and shouldn't change anything.